A1/e-meix-WordCount.java

```diff
@@ -0,0 +1,113 @@
+ */
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import org.apache.commons.cli.*;
+
+public class e_meix_WordCount {
+	/*
+	 * Checking whether a given file object exists and is a readable file. 
+	 * If that is the case true is returned, otherwise execution is terminated with an error message. 
+	 */
+	public static boolean checkInfile(File file) {
+		if(!(file.exists())) {
+			System.out.println("Error: invalid arguments - file " + file.getPath() + " does not exist.");
+			System.exit(1);
+		}
+		if(!(file.isFile())) {
+			System.out.println("Error: invalid arguments - " + file.getPath() + " is not a file.");
+			System.exit(1);
+		}
+		if(!(file.canRead())) {
+			System.out.println("Error: invalid arguments - file " + file.getPath() + " not readable.");
+			System.exit(1);
+		}
+		return true;
+	}
+	
+	/*
+	 * Takes hashmap and returns list of entries sorted first by values then by keys
+	 */
+	public static List<HashMap.Entry<String, Integer>> sortByValueThenKey(HashMap<String, Integer> hm) {
+		List<HashMap.Entry<String, Integer>> list = new ArrayList<HashMap.Entry<String, Integer>>(hm.entrySet());
+		Collections.sort(list, new Comparator<HashMap.Entry<String, Integer>>() {
+			public int compare(HashMap.Entry<String, Integer> e1, HashMap.Entry<String, Integer>e2) {
+				if(e1.getValue()>e2.getValue()) return -1;
+				if(e1.getValue()<e2.getValue()) return 1;
+				return e1.getKey().compareTo(e2.getKey());	
+			}
+		});
+		return list;
+	}
+
+	public static void main(String[] args) throws ParseException {
+		// defining possible flags
+		Options opt = new Options();
+		opt.addOption("I", false, "Ignore case.");
+		opt.addOption("l", false, "Output sorted list and frequencies of all words contained in input file.");
+		
+		// parsing command line arguments:
+		CommandLineParser p = new DefaultParser();
+		CommandLine cmd = p.parse(opt,args);
+		
+		if(cmd.getArgs().length < 1) {
+		    System.out.println("Error: Missing argument - file name.");
+		    System.exit(1);
+		}
+		if(cmd.getArgs().length > 1) {
+			System.out.println("Error: too many arguments.");
+			System.exit(1);
+		}
+		
+		File inFile = new File(cmd.getArgs()[0]);
+		checkInfile(inFile);
+		
+		HashMap<String, Integer> wordsFreq = new HashMap<String, Integer>();
+		Set<String> words = new HashSet<String>();
+		
+		// read input file word by word
+		int wordsTotal = 0;
+		Scanner in;
+		try {
+			in = new Scanner(inFile);
+			in.useDelimiter("[^\\p{L}\\p{No}]+");  
+			if(cmd.hasOption("l")){
+				while(in.hasNext()){
+					String w;
+					if(cmd.hasOption("I")) w=in.next().toLowerCase();
+					else w = in.next();
+					Integer i = wordsFreq.get(w);
+				    if(i==null) wordsFreq.put(w, 1);
+				    else wordsFreq.put(w, ++i);
```

Comment:
somewhat more clearly
```suggestion
				    else wordsFreq.put(w, i+1);
```